lib/manageiq/providers/workflows/runner.rb

```diff
@@ -0,0 +1,86 @@
+module ManageIQ
+  module Providers
+    module Workflows
+      class Runner
+        include Vmdb::Logging
+
+        class << self
+          def runner
+            @runner ||= new.tap(&:start)
+          end
+        end
+
+        attr_reader :workflows
+
+        def initialize
+          require "floe"
+          require "concurrent/hash"
+
+          @workflows          = Concurrent::Hash.new
+          @docker_wait_thread = nil
+        end
+
+        def start
+          $workflows_log.debug("Runner: Starting workflows runner...")
+          self.docker_wait_thread = Thread.new { docker_wait }
+          $workflows_log.debug("Runner: Starting workflows runner...Complete")
+        end
+
+        def stop
+          $workflows_log.debug("Runner: Stopping workflows runner...")
+          stop_thread(docker_wait_thread)
+
+          self.docker_wait_thread = nil
+          $workflows_log.debug("Runner: Stopping workflows runner...Complete")
+        end
+
+        def add_workflow(workflow, queue_args)
+          workflows[workflow.id] ||= [workflow, queue_args]
```

Comment:
also, threw around the idea of passing a lambda instead of queue_args.

```ruby
-> { workflow.queue(queue_args] }
```

This will allow us to call with a Workflow instead of a `ScriptRunner`

Reply:
> I'm feeling like we should register by the container_ref

The container_ref changes constantly I don't think that we can index off of that, we'd be re-indexing every single lookup.